pkg/transport/generic_transport_controller.go

```diff
@@ -718,54 +718,56 @@ func (c *genericTransportController) updateWrappedObjectsAndFinalizer(ctx contex
+		object, err := c.wdsDynamicClient.Resource(gvr).Get(ctx, clause.Name, metav1.GetOptions{})
 		if err != nil {
-			return nil, groupResources, fmt.Errorf("failed to get required cluster-scoped object '%s' with gvr %s from WDS - %w", clusterScopedObject.Name, gvr, err)
+			return nil, groupResources, fmt.Errorf("failed to get required cluster-scoped object '%s' with gvr %s from WDS - %w", clause.Name, gvr, err)
 		}
-		gr := metav1.GroupResource{Group: clusterScopedObject.GroupVersionResource.Group, Resource: clusterScopedObject.GroupVersionResource.Resource}
+		gr := metav1.GroupResource{Group: clause.GroupVersionResource.Group, Resource: clause.GroupVersionResource.Resource}
 		groupResources.Insert(gr)
-		objectsToPropagate = append(objectsToPropagate, TransformObject(ctx, c.customTransformCollection, gr, object, binding.Name))
+		wrapees = append(wrapees, Wrapee{TransformObject(ctx, c.customTransformCollection, gr, object, binding.Name), clause.CreateOnly})
 	}
 	// add namespace-scoped objects to the 'objectsToPropagate' slice
-	for _, namespaceScopedObject := range binding.Spec.Workload.NamespaceScope {
-		gvr := schema.GroupVersionResource(namespaceScopedObject.GroupVersionResource)
-		object, err := c.wdsDynamicClient.Resource(gvr).Namespace(namespaceScopedObject.Namespace).Get(ctx, namespaceScopedObject.Name, metav1.GetOptions{})
+	for _, clause := range binding.Spec.Workload.NamespaceScope {
+		gvr := schema.GroupVersionResource(clause.GroupVersionResource)
+		object, err := c.wdsDynamicClient.Resource(gvr).Namespace(clause.Namespace).Get(ctx, clause.Name, metav1.GetOptions{})
 		if err != nil {
-			return nil, groupResources, fmt.Errorf("failed to get required namespace-scoped object '%s' in namespace '%s' with gvr '%s' from WDS - %w", namespaceScopedObject.Name,
-				namespaceScopedObject.Namespace, gvr, err)
+			return nil, groupResources, fmt.Errorf("failed to get required namespace-scoped object '%s' in namespace '%s' with gvr '%s' from WDS - %w", clause.Name,
+				clause.Namespace, gvr, err)
 		}
-		gr := metav1.GroupResource{Group: namespaceScopedObject.GroupVersionResource.Group, Resource: namespaceScopedObject.GroupVersionResource.Resource}
+		gr := metav1.GroupResource{Group: clause.GroupVersionResource.Group, Resource: clause.GroupVersionResource.Resource}
 		groupResources.Insert(gr)
-		objectsToPropagate = append(objectsToPropagate, TransformObject(ctx, c.customTransformCollection, gr, object, binding.Name))
+		wrapees = append(wrapees, Wrapee{TransformObject(ctx, c.customTransformCollection, gr, object, binding.Name), clause.CreateOnly})
 	}
 
-	return objectsToPropagate, groupResources, nil
+	return wrapees, groupResources, nil
 }
 
 // computeDestToWrappedObjects returns the following three things.
-//   - the destToWrappedObject function. This maps a destination to the customized wrapped object
+//   - the destToWrappedObject function. This maps a destination to the slice of wrapped customized objects
```

Comment:
s/customized/possibly customized/.

Reply:
I am thinking "certainly as customized as the user is calling for". But that's a little too long to spell out in the name.